Nj persistance with json

nirjharm opened the pull request.

bowu commented on libquery/src/OrderByOp.cpp:
```diff
@@ -0,0 +1,94 @@
+#include <arrow/compute/api_vector.h>
+
+// TODO (bowu): this implementation works if the associated variable of the
+// order by key is projected (e.g., RETURN a.firstName order by a.firstName does
```
Great question. I fixed the problem.

e-mcginnis committed:
EntityTypeID management in PG and libtsuba

Co-authored-by: Evan McGinnis <evan@katanagraph.com>
Co-authored-by: Giorgi Lomia <lomiag@berea.edu>"

amberhassaan committed:
fix build: restore method lost in rebase

amberhassaan committed:
another build fix

amberhassaan committed:
Fix build

GioLomia committed:
Putting back the constructors missing during rebasae.

GioLomia committed:
Revert "Putting back the constructors missing during rebasae."

This reverts commit 59bbbc5f8082c68d513407a2afeaa1f972571049.

amberhassaan committed:
Fix python wrapper for PropertyGraph

e-mcginnis committed:
fix double open of rdg_file in PG::Make

fixes memory leaks in lonestar tests

fix formatting

GioLomia committed:
Started scenario 1 test

Format fix

e-mcginnis committed:
Add Equal and Diff functions for EntityTypeManager

update PG:Equals and PG:Diff functions to include new type storeage
structures

make EntityTypeManager Getters const

add PG entity_type_manager getters

add PG function to get id_arrays size

ensure topology and node/edge type ids agree when loading a graph

e-mcginnis committed:
add TESTING ONLY header to Equals/Diff functions

further work on test

e-mcginnis committed:
actually fix python build

Commit:
Added the scenario 1 test

e-mcginnis committed:
fix bug where properties are loaded into RDG

- fix diff test to match new diff messages
- fix double call to reportDiff in PG

Commit:
Working on passing the test for scenario 1

e-mcginnis committed:
correct merge conflict resolution

Commit:
Reporting the difference between two graphs in scenario 1.

e-mcginnis committed:
one more crack at fixing the python build

e-mcginnis committed:
Fix EntityTypeIDManager Init off by one

When initializing the EntityTypeIDManager we added sets for id = 0
aka kUnknownType, but we didn't properly initialize the set

the set for EntityTypeID 0 should contain itself

Added a name for kUnknownType so we can just use the proper
AddAtomicEntityType and avoid these kind of bugs

e-mcginnis committed:
add additional checks to EntityTypeManager::Init

e-mcginnis committed:
fix node/edge swap when storing as json

Commit:
JSON Side

Commit:
Format
